dss/stepfunctions/s3copyclient/implementation.py

```diff
@@ -181,6 +190,15 @@ def calculate_range_for_part(self, part_id) -> typing.Tuple[int, int]:
         return result
 
 
+def _set_uncached_tag(destination_bucket: str, destination_key: str):
+    client = boto3.client("s3")
```

Comment:
This will create a new boto3 s3 client every time `_set_uncached_tag` is called.

Here and everywhere else, please use cached client via the import `from dss.util.aws.clients import s3`